File: python/linkedlist.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
    
class LinkedList:
    def __init__(self):
        self.head = None
        
    def printlist(self):
        temp = self.head
        while(temp):
            print(temp.data)
            temp = temp.next
    
    def addNode(self, data):
        if(self.head == None):
            self.head = Node(data)
            return
        temp = self.head
        while(temp.next):
            temp = temp.next
            
        temp.next = Node(data)
          
if __name__=='__main__':
    llist = LinkedList()
    llist.addNode(1)
    llist.addNode(2)
    llist.addNode(3)
    llist.addNode(4)
    llist.addNode(5)
    
    llist.printlist()

File: Java/src/com/maddy/collections/LinkedList.java
package com.maddy.collections;

/**
 * Created by madhukar.b on 16/10/16.
 */
public class LinkedList
{
    private UniDirectionalNode head;
    public LinkedList()
    {
        this.head = null;
    }

    /**
     * Add element to list at the end
     * @param data element to be inserted
     */
    public void add(int data)
    {
        if(head == null)
        {
            head = new UniDirectionalNode(data);
            return;
        }

        UniDirectionalNode temp = head;
        while(temp.next != null)
        {
            temp = temp.next;
        }

        temp.next = new UniDirectionalNode(data);
    }

    public void addAtBeginning(int data)
    {
        UniDirectionalNode temp = new UniDirectionalNode(data);
        temp.next = head;
        head = temp;
    }

    public void print()
    {
        UniDirectionalNode temp = head;
        while(temp != null)
        {
            System.out.print(temp.data + " ");
            temp = temp.next;
        }

        System.out.println();
    }
}
